Modules/peg_parser.c

```diff
@@ -2,22 +2,33 @@
 #include <pegen_interface.h>
 
 PyObject *
-_Py_parse_file(PyObject *self, PyObject *args)
+_Py_parse_file(PyObject *self, PyObject *args, PyObject *kwds)
 {
+    static char *keywords[] = {"file", "mode", NULL};
     char *filename;
+    char *mode_str = "exec";
 
-    if (!PyArg_ParseTuple(args, "s", &filename)) {
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "s|s", keywords, &filename, &mode_str)) {
         return NULL;
     }
 
+    int mode;
+    if (strcmp(mode_str, "exec") == 0) {
+        mode = Py_file_input;
+    } else if (strcmp(mode_str, "eval") == 0) {
+        mode = Py_eval_input;
+    } else {
+        return PyErr_Format(PyExc_ValueError, "mode must be either 'exec' or 'eval'");
+    }
```

Comment:
If you're parsing from a file, why is "eval" needed?

Reply:
It's not really needed, but should we disallow this?

Reply:
I see no use.

Reply:
I see no use.